src/types.ts

```diff
@@ -0,0 +1,84 @@
+export type JavaScriptDataType =
+  | number
+  | string
+  | boolean
+  | null
+  | undefined
+  | symbol
+  | bigint
+  | object;
+
+export type ShaderTypes = "VERTEX_SHADER" | "FRAGMENT_SHADER";
+
+export type Shader = { vertex: string; fragment: string };
+
+export interface ShaderProgram {
+  program: WebGLProgram;
+  attr: {
+    bufferType: GLenum;
+    buffer: WebGLBuffer;
+    drawType: GLenum;
+    valueType: GLenum;
+    size: GLint;
+    attribute: number;
+    data: Float32Array;
+  }[];
+  uniform: {
+    [key: string]: WebGLUniformLocation | null;
+  };
+}
+
+export type Color = [number, number, number, number?];
+
+export interface GradientElement {
+  color: Color;
+  offset: number;
+  value: any;
+}
+
+export interface MappedGradient {
+  value: Float32Array;
+  length: number;
+  offset: number[];
+}
+
+export interface Point {
+  x: number;
+  y: number;
+  value: number;
+}
+
+// TODO need to check optional
+export type HeatmapConfig = {
+  size?: number;
+  max?: number;
+  min?: number;
+  intensity?: number;
+  translate?: [number, number];
+  zoom?: number;
+  angle?: number;
+  opacity?: number;
+  gradient: GradientElement[];
+  backgroundImage: BackgroundImageConfig;
+};
+
+export type HearmapExData = {
+  posVec: Float32Array;
+  rVec: Float32Array;
+  minMax: {
+    min: number;
+    max: number;
+  };
+};
+
+// TODO need to check optional
+export interface BackgroundImageConfig {
+  url?: string;
```

Comment:
Would the user of the library always need to pass both url and image? Or it can be either url or image ? Or both image and url are optional to be passed?

Reply:
Heatmap supports just 1 background(Optional), which will be either Image path URL or Image Blob.